javatests/arcs/core/data/proto/SchemaProtoTest.kt

```diff
@@ -173,4 +181,89 @@ class SchemaProtoDecoderTest {
             )
         )
     }
+
+    @Test
+    fun encode_schema() {
```

Comment:
I wonder if there is a cleaner way to share the proto examples between the encoding and decoding tests? That way we can ensure the tests are comprehensive for both encoding and decoding. e.g., I just noticed that the decoding tests do not cover entities with collection fields at all.

Reply:
Good idea. I've switched this to use roundTrip tests, where you encode and decode and check it looks the same. It's *much* tidier now.